src/HearThis/Communication/AndroidSynchronization.cs

```diff
@@ -47,20 +48,40 @@ public static void DoAndroidSync(Project project)
 			dlg.ShowAndroidIpAddress(); // AFTER we set our IP address, which may be used to provide a default
 			dlg.GotSync += (o, args) =>
 			{
-				var theirLink = new AndroidLink();
-				// Enhance: some way to validate that we really got an IP address.
-				theirLink.AndroidAddress = AndroidSyncDialog.AndroidIpAddress;
-				var ourLink = new WindowsLink(Program.ApplicationDataBaseFolder);
-				var merger = new RepoMerger(project, ourLink, theirLink);
-				merger.Merge(dlg.ProgressBox);
-				//Update info.txt on Android
-				var infoFilePath = project.GetProjectRecordingStatusInfoFilePath();
-				File.WriteAllText(infoFilePath, project.GetProjectRecordingStatusInfoFileContent());
-				var theirInfoTxtPath = project.Name + "/" + Project.InfoTxtFileName;
-				theirLink.PutFile(theirInfoTxtPath, File.ReadAllBytes(infoFilePath));
-				theirLink.SendNotification("syncCompleted");
-				dlg.ProgressBox.WriteMessage("Sync completed successfully");
-				//dlg.Close();
+				try
+				{
+					var theirLink = new AndroidLink();
+					// Enhance: some way to validate that we really got an IP address.
+					theirLink.AndroidAddress = AndroidSyncDialog.AndroidIpAddress;
+					var ourLink = new WindowsLink(Program.ApplicationDataBaseFolder);
+					var merger = new RepoMerger(project, ourLink, theirLink);
+					merger.Merge(dlg.ProgressBox);
+					//Update info.txt on Android
+					var infoFilePath = project.GetProjectRecordingStatusInfoFilePath();
+					File.WriteAllText(infoFilePath, project.GetProjectRecordingStatusInfoFileContent());
+					var theirInfoTxtPath = project.Name + "/" + Project.InfoTxtFileName;
+					theirLink.PutFile(theirInfoTxtPath, File.ReadAllBytes(infoFilePath));
+					theirLink.SendNotification("syncCompleted");
+					dlg.ProgressBox.WriteMessage("Sync completed successfully");
+					//dlg.Close();
+				}
+				catch (WebException ex)
+				{
+					if (ex.Status == WebExceptionStatus.NameResolutionFailure)
+					{
+						dlg.ProgressBox.WriteError("HearThis could not make sense of the address you gave for the device. Please try again.");
+					} else if (ex.Status == WebExceptionStatus.ConnectFailure)
+					{
+						dlg.ProgressBox.WriteError("HearThis could not connect to the device. Are you sure you are on the same WiFi network? Could there be a firewall blocking things?");
+					} else if (ex.Status == WebExceptionStatus.ConnectionClosed)
+					{
+						dlg.ProgressBox.WriteError("The connection to the device closed unexpectedly. Maybe you tried to do something else with it, or it went to sleep? Please try again.");
+					}
+					else
+					{
+						dlg.ProgressBox.WriteError("Something went wrong with the transfer. The system message is {0}. Please try again, or report the problem if it keeps happening", ex.Message);
```

Comment:
Probably need to give some kind of indication where, how, or to whom they should report the problem:
Please try again. If the problem persists, on the More menu, click Support to go to the community site, and then report the problem there by clicking "Ask a Question" next to the HearThis icon.